src/resolvers/user.resolver.ts

```diff
@@ -25,6 +26,29 @@ export const userResolver = {
         return undefined;
       }
     },
+
+    login: async function (_: never, { credentials }: { credentials: { email: string; password: string } }) {
+      const matchingUser = await getDataSource().manager.findOne(User, { where: { email: credentials.email } });
+
+      if (!matchingUser) {
+        throw new InvalidLoginCredentials();
+      }
+
+      const passwordMatch = await checkPassword(credentials.password, matchingUser.password);
+      if (!passwordMatch) {
+        throw new InvalidLoginCredentials();
+      }
+
+      return {
+        user: {
+          id: matchingUser.id,
+          name: matchingUser.name,
+          email: matchingUser.email,
+          birthDate: matchingUser.birthDate,
+        },
```

Comment:
Aqui se vc retornar o matchingUser, o GraphQL cuida de filtrar os campos certos, aí vc n precisa repetir linha a linha.

Reply:
interessante! dessa eu não sabia
mas aí na hora de gerar o token eu não poderia usar o `matchingUser`, né?

Reply:
Seria só pra esse campo `user` que eu to comentando. Vc pode reaproveitar se vc montar um objeto do tipo:

```ts
const response = { user: matchingUser, token: yourToken };
```

Reply:
justo! bem melhor msm